test/cpp/end2end/async_end2end_test.cc

```diff
@@ -197,6 +197,24 @@ class Verifier {
   bool spin_;
 };
 
+class ServerBuilderSyncPluginDisabler : public ServerBuilderOption {
+ public:
+  void UpdateArguments(ChannelArguments* arg) GRPC_OVERRIDE {}
+
+  void UpdatePlugins(
+      std::map<grpc::string, std::unique_ptr<ServerBuilderPlugin> >*
+          plugins) GRPC_OVERRIDE {
+    auto plugin = plugins->begin();
+    while (plugin != plugins->end()) {
+      if ((*plugin).second->has_synchronous_methods()) {
+        plugins->erase(plugin++);
```

Comment:
The iterator will not be invalidated. erase will clear element pointed by `plugin`, while `plugin` becomes `plugin + 1`.